libafl/src/bolts/serdeany.rs

```diff
@@ -35,6 +35,8 @@ pub trait SerdeAny: Any + erased_serde::Serialize {
     fn as_any(&self) -> &dyn Any;
     /// returns this as mutable Any trait
     fn as_any_mut(&mut self) -> &mut dyn Any;
+    /// returns this as boxed Any trait
+    fn as_any_boxed(self: Box<Self>) -> Box<dyn Any>;
```

Comment:
Can't you impl `SerdeAny` for `Box<xyz>` or something like that? Does it really need its own method?

Reply:
> Can't you impl `SerdeAny` for `Box<xyz>` or something like that? Does it really need its own method?

not without losing the box.

I needed this implementation for the serdeanymap::remove that I implemented, but I don't actually use it anymore. I can remove it again. 

I ended up with this solution after reading this article: https://articles.bchlr.de/traits-dynamic-dispatch-upcasting . I just tried factoring this method out into an extension trait, but couldn't get it to work. it may be possible, but I'd rather not spend more time on this type system shenanigans. what speaks against the method though? its only implemented via macros anyways, so the user doesn't need to care.